### PR TITLE
Update Llama.cpp & Llama 3 Support

### PR DESCRIPTION
**Pull request by PSchmiedmayer**
# Update Llama.cpp & Llama 3 Support

## :gear: Release Notes 
- Updates llama.cpp
- Supports Llama 3

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PSchmiedmayer** [comment]
@vishnuravi @philippzagar Feel free to add to this WIP here after lifting llama.cpp. @vishnuravi can you share some of the issues you had with Llama3 so we can document this here in case this gets stale or someone else wants to pick it up?

**Comment by vishnuravi** [comment]
> @vishnuravi @philippzagar Feel free to add to this WIP here after lifting llama.cpp. @vishnuravi can you share some of the issues you had with Llama3 so we can document this here in case this gets stale or someone else wants to pick it up?

Hi @PSchmiedmayer, I'm running into the same issue documented here: https://github.com/ollama/ollama/issues/3759

**Review by vishnuravi — APPROVED** [review]

**Comment by vishnuravi** [comment]
> > @vishnuravi @philippzagar Feel free to add to this WIP here after lifting llama.cpp. @vishnuravi can you share some of the issues you had with Llama3 so we can document this here in case this gets stale or someone else wants to pick it up?
> 
> Hi @PSchmiedmayer, I'm running into the same issue documented here: [ollama/ollama#3759](https://github.com/ollama/ollama/issues/3759) and here: https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct/discussions/4. The model does not stop at the stop token and continues generating indefinitely.

I was able to resolve this issue by using a different GGUF with the end token correctly configured.

File: Sources/SpeziLLMLocal/LLMLocalSchema+PromptFormatting.swift
Comment (PSchmiedmayer) [inline Sources/SpeziLLMLocal/LLMLocalSchema+PromptFormatting.swift:1]: Thank you @vishnuravi for the updates here!

**Pull request merged**